server/src/internalClusterTest/java/org/opensearch/snapshots/SearchableSnapshotIT.java

```diff
@@ -118,6 +125,209 @@ public void testCreateSearchableSnapshot() throws Exception {
         assertIndexDirectoryDoesNotExist("test-idx-1-copy", "test-idx-2-copy");
     }
 
+    /**
+     * Tests the functionality of {@link org.opensearch.cluster.routing.allocation.allocator.RemoteShardsBalancer} to
+     * ensure it can assign remote shards to a node with local shards given it has the
+     * {@link org.opensearch.cluster.node.DiscoveryNodeRole#SEARCH_ROLE} capabilities.
+     */
+    public void testSearchableSnapshotAllocationForLocalAndRemoteShardsOnSameNode() throws Exception {
+        final int numReplicasIndex = randomIntBetween(1, 4);
+        final String indexName = "test-idx";
+        final String restoredIndexName = indexName + "-copy";
+        internalCluster().ensureAtLeastNumSearchAndDataNodes(numReplicasIndex + 1);
+        final Client client = client();
```

Comment:
Is it possible to create some sort of "create searchable snapshot index" helper method that will do all this similar stuff shared between tests?

Reply:
Done. Should be a lot less verbose now.